src/Algebra/Apartness/Properties/HeytingCommutativeRing.agda

```diff
@@ -11,23 +11,123 @@ open import Algebra.Apartness.Bundles using (HeytingCommutativeRing)
 module Algebra.Apartness.Properties.HeytingCommutativeRing
   {c ℓ₁ ℓ₂} (HCR : HeytingCommutativeRing c ℓ₁ ℓ₂) where
 
-open import Data.Product using (_,_; proj₂)
-open import Algebra using (CommutativeRing; RightIdentity)
+open import Function using (_∘_)
+open import Data.Product using (_,_; proj₁; proj₂)
+open import Algebra using (CommutativeRing; RightIdentity; Invertible; LeftInvertible; RightInvertible)
 
 open HeytingCommutativeRing HCR
-open CommutativeRing commutativeRing using (ring)
+open CommutativeRing commutativeRing using (ring; *-commutativeMonoid)
 
-open import Algebra.Properties.Ring ring using (-0#≈0#)
+open import Algebra.Properties.Ring ring
+  using (-0#≈0#; -‿distribˡ-*; -‿distribʳ-*; -‿anti-homo-+; -‿involutive)
+open import Relation.Binary.Definitions using (Symmetric)
+import Relation.Binary.Reasoning.Setoid as ReasonSetoid
+open import Algebra.Properties.CommutativeMonoid *-commutativeMonoid
 
+private variable
+  x y z : Carrier
+
+leftInvertible⇒# : LeftInvertible _≈_ 1# _*_ (x - y) → x # y
+leftInvertible⇒# = invertible⇒# ∘ leftInv→Inv
+
+rightInvertible⇒# : RightInvertible _≈_ 1# _*_ (x - y) → x # y
+rightInvertible⇒# = invertible⇒# ∘ rightInv→Inv
 
 x-0≈x : RightIdentity _≈_ 0# _-_
 x-0≈x x = trans (+-congˡ -0#≈0#) (+-identityʳ x)
 
 1#0 : 1# # 0#
-1#0 = invertible⇒# (1# , 1*[x-0]≈x , [x-0]*1≈x)
+1#0 = leftInvertible⇒# (1# , 1*[x-0]≈x)
   where
-  1*[x-0]≈x : ∀ {x} → 1# * (x - 0#) ≈ x
+  1*[x-0]≈x : 1# * (x - 0#) ≈ x
   1*[x-0]≈x {x} = trans (*-identityˡ (x - 0#)) (x-0≈x x)
 
-  [x-0]*1≈x : ∀ {x} → (x - 0#) * 1# ≈ x
-  [x-0]*1≈x {x} = trans (*-comm (x - 0#) 1#) 1*[x-0]≈x
+x#0y#0→xy#0 : x # 0# → y # 0# → x * y # 0#
+x#0y#0→xy#0 {x} {y} x#0 y#0 = leftInvertible⇒# (y⁻¹ * x⁻¹ , y⁻¹*x⁻¹*x*y≈1)
+  where
+  open ReasonSetoid setoid
+
+  InvX : Invertible _≈_ 1# _*_ (x - 0#)
+  InvX = #⇒invertible x#0
+
+  x⁻¹ = InvX .proj₁
+
+  x⁻¹*x≈1 : x⁻¹ * (x - 0#) ≈ 1#
+  x⁻¹*x≈1 = InvX .proj₂ .proj₁
+
+  x*x⁻¹≈1 : (x - 0#) * x⁻¹ ≈ 1#
+  x*x⁻¹≈1 = InvX .proj₂ .proj₂
+
+  InvY : Invertible _≈_ 1# _*_ (y - 0#)
+  InvY = #⇒invertible y#0
```

Comment:
I feel like a pattern matching here (and elsewhere) would be better e.g. `(y⁻¹ , y⁻¹*y≈1 , y*y⁻¹≈1) = #⇒invertible y#0` would save a lot of lines?